test/test_macros.jl

```diff
@@ -1894,4 +1894,35 @@ end
 
 #!format: on
 
+function test_matrix_in_vector_set()
+    model = Model()
+    @variable(model, X[1:2, 1:2])
+    A = [1 2; 3 4]
+    @test_throws_strip(
+        ErrorException(
+            "In `@constraint(model, X >= A)`: Unsupported matrix in " *
+            "vector-valued set. Did you mean to use the broadcasting syntax " *
+            "`.>=` instead?",
+        ),
+        @constraint(model, X >= A),
+    )
+    @test_throws_strip(
+        ErrorException(
+            "In `@constraint(model, X <= A)`: Unsupported matrix in " *
+            "vector-valued set. Did you mean to use the broadcasting syntax " *
+            "`.<=` instead?",
+        ),
+        @constraint(model, X <= A),
+    )
+    @test_throws_strip(
+        ErrorException(
+            "In `@constraint(model, X == A)`: Unsupported matrix in " *
+            "vector-valued set. Did you mean to use the broadcasting syntax " *
+            "`.==` instead?",
+        ),
+        @constraint(model, X == A),
```

Comment:
We recently added (https://github.com/jump-dev/JuMP.jl/pull/3273) support for `vector == vector`. But it is less clear what to do for `matrix == matrix`. 

It gets lowered to `X - A in Zeros()`, but `X - A` is a matrix, and `Zeros()` is a vector-valued set, which triggers this error.

Reply:
We could make the equality case work.
The inequality, it's better that it's an error since the user most probably forgot `PSDCone()`

Reply:
I think it wouldn't even be possible to support inequalities, as JuMP cannot know whether the user wants PSDCone() or HermitianPSDCone().

In any case, I concur, allowing inequalities leads to ambiguities. Maybe the user has a symmetric matrix, but wants to constrain it to be elementwise nonnegative.

There's also a bug that always happens with YALMIP: the user has a matrix they think is Hermitian but it's actually not because of numerical noise. Then the user writes >= 0 thinking it's constraining it to be PSD, but YALMIP instead interprets that as elementwise non-negativity.

Reply:
> There's also a bug that always happens with YALMIP: the user has a matrix they think is Hermitian but it's actually not because of numerical noise. Then the user writes >= 0 thinking it's constraining it to be PSD, but YALMIP instead interprets that as elementwise non-negativity.

Indeed, we want to avoid that, an error asking the user to be explicit about it is better than trying to save a few keystrokes

Reply:
Let's merge this and see how we go re-feedback. The non-symmetric case `X .== Y` is pretty easy to type, and there's a nice error message.

The specialized support for symmetric and hermitian is more useful because of the redundant constraints.

Agree on inequalities.

Reply:
> The case of equalities is not so clear-cut because the user intention is not ambiguous. If JuMP interprets == as .== in the non-symmetric case the result will be a slowdown, not a bug (assuming the user thought the matrix was Hermitian).

> I would just change the error message to say "Please declare your matrix as Symmetric or Hermitian if it is the case. Otherwise use .=="

We can't really do that since JuMP is a general purpose solver so having equality between matrices is completely fine and it's weird to talk about `Symmetric` or `Hermitian` matrices outside the context of SDP.

Reply:
Fair enough, but the error message seems to imply that == is never supported for matrices, which is not the case.

Reply:
Good point,  `JuMP.Zeros` is indeed not a vector-valued set (since it works with scalars, Symmetric and Hermitian). "Equality between `$(typeof(matrix))`. Use `.==` for elementwise inequality or wrap the matrix in  `Symmetric` or `Hermitian`) for equality between matrices of such special structure"

Reply:
Perfect, thanks.

Reply:
Updated the docs and the error message.